### PR TITLE
adapt to new api

### LINKED ISSUE #145: TaskGenerator,Sample
I have two questions. 1. Is the task generator API cancelled now? 2. How can I sample supportset and queryset from a sample task, I mean by which API?

### PR DESCRIPTION
**Pull request by joemzhao**
### Description

Fixes #145 

update the text example with the new api. since ``benchmarks`` for text is not available yet so this pr is a bit verbose.

### Contribution Checklist

If your contribution modifies code in the core library (not docs, tests, or examples), please fill the following checklist.

- [x] My contribution modifies code in the main library.
- [x] My modifications are tested.
- [ ] My modifications are documented.

#### Optional

If you make major changes to the core library, please run `make alltests` and copy-paste the content of `alltests.txt` below.

~~~shell
[PASTE HERE]
~~~

### CODE REVIEW

**Comment by seba-1511** [comment]
Hi @joemzhao, thanks for the PR -- it looks good and I'll merge it as soon as #151 gets in.

Have you tried to run this example? What kind of results do you get? I had plans to remove the `l2l.text` submodule, as we haven't been able to get good results with it.

**Comment by joemzhao** [comment]
> Hi @joemzhao, thanks for the PR -- it looks good and I'll merge it as soon as #151 gets in.

Hi @seba-1511 , thanks for your reply! Yes I guess the features added by #151 are more important and definitely have higher priority -- I will update this merge accordingly after #151.

> Have you tried to run this example? What kind of results do you get? 

Yes I ran some small tests. So for a 5-way 1-shot problem with iterations 50 and fas 5, the best validation accuracy is 0.2750, which is better than the majority baseline, but surely not a good score compared with vision tasks. 

>I had plans to remove the `l2l.text` submodule, as we haven't been able to get good results with it.

 I am actually expecting the bad results of script ``news_topic_classification.py`` for nlp tasks, since:
- the features are not as transferable as vision tasks, which surely hurt the quick adaptation (c.f.https://openreview.net/forum?id=H1emfT4twB)
- roberta in this setup is not finetuned. finetuning it should improve the results significantly.

I am feeling it's a pity to remove `l2l.text` since I am feeling meta-learn should still help nlp, e.g., for different languages. I am working on a related project, and planning to start with `learn2learn` (thanks again for this repo ;)). so perhaps I could try to extend this section a bit? Ideally I should be able to extend it like the vision benchmark in the `examples` folder.

**Comment by seba-1511** [comment]
Thanks for the details, I'll merge it now.

Since you have a use case for `l2l.text`, we can keep it in the library. Contributions to the text submodule would be very welcome. Feel free to get in touch on our [slack](http://slack.learn2learn.net/) if you need help with that.
